### DISCUSSION TITLE
Slow memory/database access

### DISCUSSION THREAD

**Discussion by otseng** [ish-app/ish#1411] [Q&A]
I'm trying to run a Python app in iSH on my iPhone 12.  

https://github.com/eliranwong/UniqueBible/wiki/Running-UBA-server-on-iOS

It appears memory access operations, in particular reading from a database, is quite slow.   Even when running "apk add python3", it was pretty slow.  

The app is running as a http server and most operations need to query a database.  When querying the database, it takes over 30 seconds.  Without querying the database, response is under 1 second.

On Android, everything is under 1 second, so it's not the app that is slow.

Anybody else experiencing slow memory access?  
Are there any tweaks to make it faster?

**Comment by ifuchs** [comment]
Can you say more about the db?

**Reply by otseng** [reply]
It's using sqlite.

But, I don't think it's just a DB issue.

Running "git clone", "pip install", "apk add" also takes a long time.

It's not a network problem either since all my devices can download fast.

**Comment by ifuchs** [comment]
If you run sqlite3 to access the db in the same way as the app, I presume it is also very slow? I can time an apk add on my iPhone 7 for comparison if you give me an example to try.

**Reply by otseng** [reply]
Good suggestion.  Tried out sqlite3 command and response time is reasonable.   Seems like it's something with Python and sqlite.  Will do more research on this.

**Comment by otseng** [comment]
Looks like many others have experienced the same issue:

https://github.com/ish-app/ish/issues/129
